src/controllers/cartController.js

```diff
@@ -0,0 +1,180 @@
+const getCart = async (req, res, next) => {
+  try {
+    const userId = req.user.userId;
+
+    const cart = await Cart.findOne({ createdBy: userId }).populate(
+      'orderItems.book'
+    );
+    if (!cart) {
+      throw new NotFoundError('Cart not found');
+    }
+
+    const updatedItems = cart.orderItems.filter(
+      (item) => item.book && item.book.isAvailable
+    );
+
+    if (updatedItems.length !== cart.orderItems.length) {
+      cart.orderItems = updatedItems;
+      cart.total = calculateCartTotal(cart);
+
+      await cart.save();
+
+      return res.status(StatusCodes.OK).json({
+        message: 'Some unavailable books were removed from your cart.',
+        cart,
+      });
+    }
+
+    res.status(StatusCodes.OK).json({ cart });
+  } catch (error) {
+    next(error);
+  }
+};
+
+// add book to catr
+const addToCart = async (req, res, next) => {
+  try {
+    const userId = req.user.userId;
+    const { bookId } = req.body;
+
+    const book = await Book.findById(bookId);
+    if (!book || !book.isAvailable) {
+      throw new BadRequestError('Book is unavailable or does not exist');
+    }
+
+    let cart = await Cart.findOne({ createdBy: userId });
+    if (!cart) {
+      cart = await Cart.create({ createdBy: userId });
+    }
+
+    const exists = cart.orderItems.some(
+      (item) => item.book.toString() === bookId
+    );
+    if (exists) throw new BadRequestError('Book is already in the cart');
+
+    cart.orderItems.push({
+      book: book._id,
+      title: book.title,
+      author: book.author,
+      coverImageUrl: book.coverImageUrl,
+      price: book.price,
+      isAvailable: book.isAvailable,
+    });
+
+    
+    cart.total = calculateCartTotal(cart);
+
+    await cart.save();
+    res.status(201).json({ message: 'Book added to cart', cart });
+  } catch (error) {
+    next(error);
+  }
+};
+
+
+// delete from cart
+const deleteFromCart = async (req, res, next) => {
+  try {
+    const userId = req.user.userId;
+    const { id: cartItemId } = req.params;
+
+    const cart = await Cart.findOne({ createdBy: userId });
+    if (!cart) {
+      throw new NotFoundError('Cart not found');
+    }
+
+    const filteredItems = cart.orderItems.filter(
+      (item) => item._id.toString() !== cartItemId
+    );
+    if (filteredItems.length === cart.orderItems.length) {
+      throw new NotFoundError('Book not found in the cart');
+    }
+
+    cart.orderItems = filteredItems;
+
+    cart.total = calculateCartTotal(cart);
+
+    await cart.save();
+    res
+      .status(StatusCodes.OK)
+      .json({ message: 'Item removed successfully', cart });
+  } catch (error) {
+    next(error);
+  }
+};
+
+const createPaymentIntent = async (req, res, next) => {
+  try {
+    const userId = req.user.userId;
+    const cart = await Cart.findOne({ createdBy: userId }).populate(
+      'orderItems.book'
+    );
+
+    if (!cart || cart.orderItems.length === 0) {
+      throw new BadRequestError('Your cart is empty');
+    }
+
```

Comment:
handle for scenario when paymentIntent has already been processed. 

if (cart.status === 'paid') {
  return res.status(StatusCodes.OK).json({ msg: 'Payment already processed', cart });
}